web/modules/custom/registration_form_custom/registration_form_custom.module

```diff
@@ -0,0 +1,21 @@
+<?php
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\registration_form_custom\Form\NewUserRegisterForm;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\weather_block\Services\UserWeatherHandler;
+
+/**
+ * Implements hook_entity_insert().
+ */
+function registration_form_custom_entity_insert(\Drupal\user\Entity\User $entity) {
```

Comment:
Why do you need a hook when you have a custom form with all of that there in the same module?